crates/bevy_asset/src/server/mod.rs

```diff
@@ -401,6 +395,51 @@ impl AssetServer {
         handle
     }
 
+    pub(crate) fn load_erased_with_meta_transform<'a, G: Send + Sync + 'static>(
+        &self,
+        path: impl Into<AssetPath<'a>>,
+        type_id: TypeId,
+        meta_transform: Option<MetaTransform>,
+        guard: G,
+    ) -> UntypedHandle {
+        let path = path.into().into_owned();
+        let mut infos = self.data.infos.write();
+        let (handle, should_load) = infos.get_or_create_path_handle_erased(
+            path.clone(),
+            type_id,
+            None,
+            HandleLoadingMode::Request,
+            meta_transform,
+        );
+
+        if should_load {
+            let task = self.spawn_load_task(handle.clone(), path, guard);
+
+            #[cfg(not(any(target_arch = "wasm32", not(feature = "multi_threaded"))))]
+            infos.pending_tasks.insert(handle.id(), task);
+
+            #[cfg(any(target_arch = "wasm32", not(feature = "multi_threaded")))]
+            task.detach();
```

Comment:
I couldn't deduplicate this block because it uses the `infos: RwLockWriteGuard` from earlier in this fn, and I don't like the idea of passing around lock guards to functions.

Reply:
Can you just take a mutable borrow to the thing it's guarding, which I believe should be `AssetInfos`? Then you can pass that around with the guard.

Reply:
Yes, but I also don't like the idea of locking, working, unlocking, then relocking again for some other work in the same function. Is this an ok tradeoff for less code duplication?

Reply:
I don't think you need to, you can just "unwrap" the `infos` like `&mut *infos`, and that'll give you `&mut AssetInfos` while still being protected by the guard. Then you can pass that borrow to the function.

Reply:
Yep you're right, I forgot you could just pass the mut ref.